elements/barcode-reader/barcode-reader.js

```diff
@@ -57,36 +61,33 @@ class BarcodeReader extends LitElement {
    */
   render() {
     return html`
-      <div id="hidden">
+      <div class="hidden">
         <div>
           <video
             muted
             autoplay
-            id="video"
+            class="video"
             playsinline="true"
             width="${this.scale}%"
             height="${this.scale}%"
           ></video>
-          <canvas id="canvas" style="display: none; float: bottom;"></canvas>
+          <canvas class="canvas" style="display: none; float: bottom;"></canvas>
         </div>
       </div>
-      <div>
+      <div class="input" .hidden="${!this.input}">
         Result: <span><input type="text" .value="${this.value}" /> </span
-        ><button id="render">Initialize scanner</button>
-      </div>
-      <div id="hidden2">
+        ></div>
+      <span>
+      <div class="hidden2" style="">
         <div class="select">
           <label for="videoSource">Video source: </label>
-          <select id="videoSource"></select>
+          <select></select>
         </div>
-        <button id="go">Scan</button>
-      </div>
+        <button class="go">Scan</button>
+      </div><button class="render">Initialize scanner</button></span>
```

Comment:
pushing into perfection territory...

textual strings should be translatable to maximize re-use across languages / countries. https://dev.to/btopro/inside-i18n-manager-1ak6 is our solution to this tricky problem which you can read the whole way this works or reverse engineer from other tags implementing the manager. It's pretty easy to do, just replacing stuff with a `${this.t.scan}` and that object value being the string `Scan` by default (for english). It opens the door to translation in future passes across the elements portfolio (or now if you want to start translating :) )